### PR TITLE
chore: Remove `url-parse` and clean up query param helpers

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Remove `url-parse` as a direct dependency because we can use native javascript to do all of this
  - The authors of https://www.npmjs.com/package/url-parse even recommend to **not use this package** in their README.md
- This is basically a follow up of https://github.com/linode/manager/pull/8810

## Major Changes 🔄
- Removes `url-parse` package
- Removes a silly usage of `ramda`
- Updates functions and tests that previously consumed `url-parse`
- Renames functions to they are more clear
- Removes helper functions that had duplicate behavior

## How to test 🧪
- Verify query param related actions are not broken. Some examples include:
  - Table ordering query params
  - Search
  - Crazy deep links like http://localhost:3000/linodes/create?type=One-Click&appID=1102906&utm_source=marketplace&utm_medium=website&utm_campaign=Passky&showInfo=true

### CODE REVIEW

**Comment by bnussman** [comment]
Please add a chageset to your PR 🥺🚨

**Comment by bnussman-akamai** [comment]
@abailly-akamai Agreed on the positional arguments point. I will try to focus more on that in future refactors.

**Review by bnussman-akamai — COMMENTED** [review]
testing pr bot

**Review by bnussman-akamai — COMMENTED** [review]
testing more, sorry

File: packages/manager/src/utilities/queryParams.ts [hunk 1]
```diff
@@ -1,34 +1,35 @@
...
  */
-export const parseQueryParams = (str: string) =>
-  Object.fromEntries(new URLSearchParams(str));
+export const getQueryParamsFromQueryString = (queryString: string) =>
+  Object.fromEntries(new URLSearchParams(queryString));
 
-export const getQueryParam = (
-  str: string,
+/**
+ * Gets the value of a single query paramter by its name
```
Comment (dwiley-akamai) [inline packages/manager/src/utilities/queryParams.ts:18]:
```suggestion
* Gets the value of a single query parameter by its name
```

File: packages/manager/src/utilities/queryParams.ts [hunk 2]
```diff
@@ -1,34 +1,35 @@
...
+ * @param queryString The search string aka the query string to parse. This must ONLY be the query string and not the entire URL.
+ * @param paramName The name or key of the paramater you want the value of
```
Comment (dwiley-akamai) [inline packages/manager/src/utilities/queryParams.ts:20]:
```suggestion
 * @param paramName The name or key of the parameter you want the value of
```

File: packages/manager/src/utilities/queryParams.ts [hunk 3]
```diff
@@ -1,34 +1,35 @@
...
+ * @param defaultValue an optional default to fallback to if the query param is not found
+ * @returns the value of the query paramater you requested
```
Comment (dwiley-akamai) [inline packages/manager/src/utilities/queryParams.ts:22]:
```suggestion
 * @returns the value of the query parameter you requested
```

**Review by dwiley-akamai — APPROVED** [review]
LGTM 🚢

**Pull request merged**